api/proto/info/info.proto

```diff
@@ -221,6 +221,12 @@ message ZInfoConfigItemStatus {
     map<string, ZInfoConfigItem> unknownConfigItems = 2;
 }
 
+message ZInfoAppInstance {
+    string uuid = 1;
+    string name = 2;
+    string domainName = 3;
+}
+
 // Base device info, as discovered by Xen (or OS on bare metal)
 message ZInfoDevice {
```

Comment:
FWIW I was inclined to ask to add state here, but then a realize with that we'd need to trigger a device info message each time the app info state changes. So not adding it means we only need to trigger when the AI is created and deleted which is less traffic.

Reply:
I was also very tempted about it.. But for my current usage - I didn't need it - and hence did not addit in.

But in another way - it is actually good to send these more granular states - debugging would be much easier. The Device / App Instance events generated in the cloud would also be much more accurate.. DownSide is - more data would be sent on low bandwidth links..

Reply:
Hmm.. Looking at the code - we are triggering an update again anyways - in Modify.. So might as well add the State.. Will add it..

Reply:
Agreed.

Reply:
Looked at the code.. I see what. you are saying.. 

1) Change in App Instance - as long as the config is there - are sent in ZInfoApp messages
     - These are sent for every change in AppInstanceStatus

2) So - deviceInfo.AppInstance is mainly useful when the App Instance is deleted.
     - To see the various states the App is going through when it is deleted.
     - We are also sending the ZInfoApp for each of these triggers - So as long as cloud retains them, we don't need to look at the info in the device.

So based on this, I am thinking - lets add the run state, but trigger device info in AppInstanceStatus handler based on the following:

1) New AppInstanceStatus created --> Trigger devInfo
2) AppInstanceConfig deleted --> Trigger devInfo for all AIS changes
3) If AIC is present --> DO NOT trigger devInfo for AIS changes ( Except (1) above )

Does this sound good?